eZ/Publish/Core/Repository/ObjectStateService.php

```diff
@@ -676,15 +676,20 @@ protected function buildCreateInputStruct($identifier, $defaultLanguageCode, $na
      * Validates input for updating object states and builds the InputStruct object.
      *
      * @param \eZ\Publish\API\Repository\Values\ObjectState\ObjectState $objectState
-     * @param string $identifier
-     * @param string $defaultLanguageCode
-     * @param string[] $names
-     * @param string[] $descriptions
+     * @param string|null $identifier
+     * @param string|null $defaultLanguageCode
+     * @param string[]|null $names
+     * @param string[]|null $descriptions
      *
      * @return \eZ\Publish\SPI\Persistence\Content\ObjectState\InputStruct
      */
-    protected function buildObjectStateUpdateInputStruct(APIObjectState $objectState, $identifier, $defaultLanguageCode, $names, $descriptions)
-    {
+    protected function buildObjectStateUpdateInputStruct(
+        APIObjectState $objectState,
+        ?string $identifier,
+        ?string $defaultLanguageCode,
+        ?array $names,
+        ?array $descriptions
+    ): InputStruct {
         $inputStruct = new InputStruct();
 
         if ($identifier !== null && (!is_string($identifier) || empty($identifier))) {
```

Comment:
same here `!is_string($identifier)`

Reply:
and in https://github.com/ezsystems/ezpublish-kernel/pull/2728/files#diff-4ee5b7b94895992b9952ce0e423aa875R701 and https://github.com/ezsystems/ezpublish-kernel/pull/2728/files#diff-4ee5b7b94895992b9952ce0e423aa875R707